### PR TITLE
separated into abstract contracts + implemented AutDAO

### PR DESCRIPTION
**Pull request by migrenaa**

### CODE REVIEW

File: contracts/AutDAO.sol [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    /// @notice Sets the initial details of the DAO
+    /// @dev all parameters are required.
+    /// @param _deployer the address of the DAOTypes.sol contract
+    /// @param _autAddr the address of the DAOTypes.sol contract
+    /// @param _market one of the 3 markets
+    /// @param _metadata url with metadata of the DAO - name, description, logo
+    /// @param _commitment minimum commitment that the DAO requires
+    constructor(
+        address _deployer,
+        address _autAddr,
```
Comment (combatdave) [inline contracts/AutDAO.sol]: Better to pass in as the type rather than as an address

File: contracts/daoUtils/abstracts/AutIDAddress.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            "Only AutID Contract can call this!"
+        );
+        _;
+    }
+
+    function getAutIDAddress() public view override returns (address) {
+        return address(_autID);
+    }
+
+    function setAutIDAddress(address autIDAddress) public override {
```
Comment (combatdave) [inline contracts/daoUtils/abstracts/AutIDAddress.sol]: Public access

File: contracts/daoUtils/abstracts/DAOCommitment.sol [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+/// @title DAOExpander
+/// @notice The extension of each DAO that integrates Aut
+/// @dev The extension of each DAO that integrates Aut
+abstract contract DAOCommitment is IDAOCommitment {
+
+    uint _commitment;
+
+    function setCommitment(uint commitment)
+        public
```
Comment (combatdave) [inline contracts/daoUtils/abstracts/DAOCommitment.sol]: Public access

File: contracts/daoUtils/abstracts/DAOMarket.sol [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+/// @title DAOExpander
+/// @notice The extension of each DAO that integrates Aut
+/// @dev The extension of each DAO that integrates Aut
+abstract contract DAOMarket is IDAOMarket {
+
+    uint _market;
+
+    function setMarket(uint market)
+        public
```
Comment (combatdave) [inline contracts/daoUtils/abstracts/DAOMarket.sol]: Public access

File: contracts/daoUtils/abstracts/DAOMetadata.sol [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+/// @title DAOExpander
+/// @notice The extension of each DAO that integrates Aut
+/// @dev The extension of each DAO that integrates Aut
+abstract contract DAOMetadata is IDAOMetadata {
+
+    string _metadata;
+
+    function setMetadataUri(string memory metadata)
+        public
```
Comment (combatdave) [inline contracts/daoUtils/abstracts/DAOMetadata.sol]: Public access

File: contracts/daoUtils/abstracts/DAOUrls.sol [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    string[] private urls;
+
+    /// @notice url ids
+    mapping(bytes32 => uint256) private urlIds;
+
+    // URLs
+    /// @notice The listed URLs are the only ones that can be used for the DAuth
+    /// @dev adds a URL in the listed ones
+    /// @param _url the URL that's going to be added
+    function addURL(string memory _url) public override virtual {
```
Comment (combatdave) [inline contracts/daoUtils/abstracts/DAOUrls.sol]: Public access

File: contracts/daoUtils/abstracts/DAOUrls.sol [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        urlIds[urlHash] = urls.length;
+        urls.push(_url);
+
+        emit UrlAdded(_url);
+    }
+
+    /// @notice The listed URLs are the only ones that can be used for the DAuth
+    /// @dev removes URL from the listed ones
+    /// @param _url the URL that's going to be removed
+    function removeURL(string memory _url) public override virtual {
```
Comment (combatdave) [inline contracts/daoUtils/abstracts/DAOUrls.sol]: Public access

**Review by combatdave — CHANGES_REQUESTED** [review]
Think these public functions need access modifiers but otherwise looks good.

File: contracts/AutDAO.sol (hunk 1 above)
Comment (migrenaa) [reply contracts/AutDAO.sol]: okie :)

**Pull request merged**